File: src/main/java/ru/arc/hooks/BetterRTPListener.java
package ru.arc.hooks;

import me.SuperRonanCraft.BetterRTP.references.customEvents.RTP_TeleportPostEvent;
import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import ru.arc.ARC;
import ru.arc.commands.arc.subcommands.RespawnOnRtpSubCommand;
import ru.arc.configs.Config;
import ru.arc.configs.ConfigManager;

public class BetterRTPListener implements Listener {

    Config config = ConfigManager.of(ARC.getInstance().getDataPath(), "misc.yml");

    @EventHandler
    public void onBetterRTPEvent(RTP_TeleportPostEvent event) {
        Player player = event.getPlayer();
        Object ifPresent = RespawnOnRtpSubCommand.INSTANCE.getPlayersForRtp().getIfPresent(player.getName());
        if (ifPresent != null) {
            RespawnOnRtpSubCommand.INSTANCE.getPlayersForRtp().invalidate(player.getName());
            Location location = event.getLocation();
            player.setRespawnLocation(location, true);
            player.sendMessage(config.componentDef("rtp-respawn.set-spawn-message", "<green>Ваша точка возрождения установлена здесь! <gray>Чтобы изменить ее, используйте команду /sethome"));
        }
    }

}

File: src/main/kotlin/ru/arc/hooks/BetterRTPListener.kt
package ru.arc.hooks

import me.SuperRonanCraft.BetterRTP.references.customEvents.RTP_TeleportPostEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import ru.arc.rtp.RtpRespawnCompletion
import ru.arc.rtp.RtpProvider
import ru.arc.metrics.MetricsModule
import ru.arc.metrics.ProductEntryPoint
import ru.arc.metrics.ProductFeature
import ru.arc.metrics.ProductOutcome

class BetterRTPListener : Listener {
    @EventHandler
    fun onBetterRTPEvent(event: RTP_TeleportPostEvent) {
        MetricsModule.recordProductOutcome(event.player, ProductOutcome.RTP_COMPLETE, ProductFeature.RTP, ProductEntryPoint.GAMEPLAY)
        RtpRespawnCompletion.complete(event.player, RtpProvider.BETTERRTP) { event.location }
    }
}
